websites/A/Autocode/presence.ts

```diff
@@ -37,22 +44,40 @@ presence.on("UpdateData", async () => {
 		else presenceData.details = "Looking for Docs";
 		presenceData.state = `${window.location.hostname}/${path[0]}`;
 		presenceData.smallImageKey = "lib";
-	} else if (pathname.includes("/p/")) {
-		const filename = document.querySelector("div.filename")
-			? document.querySelector("div.filename").textContent.split("/").pop()
-			: null;
-		if (!filename) return;
-		const extension = filename.match(/\.\w+/g)
-			? filename.match(/\.\w+/g)[0].replace(".", "")
-			: false;
-		presenceData.details = `Project: ${path[2]}`;
-		presenceData.state = `Editing: ${filename}`;
-		presenceData.smallImageKey = "autocode";
-		if (extension && supportedLanguages.includes(extension))
-			presenceData.largeImageKey = `lang-${extension}`;
-		else presenceData.largeImageKey = "autocode";
+	} else if (pathname.includes("/mp/")) {
+		const filename = document
+			.querySelector("div.filename > [data-filename]")
+			?.textContent?.split("/")
+			?.pop();
+
+		if (filename) {
+			const extension = filename.match(/\.\w+/g)?.[0]?.replace(".", ""),
+				[line, column] = document
+					.querySelector("div.filename > [data-cursor]")
+					.textContent.split(":"),
+				replaceTemplate = (str: string) => {
+					if (str === "{0}") return;
+					else {
+						return str
+							.replace("%project%", path[2])
+							.replace("%file%", filename)
+							.replace("%line%", line)
+							.replace("%column%", column);
+					}
+				};
+
+			presenceData.details = replaceTemplate(details);
+			presenceData.state = replaceTemplate(state);
+			presenceData.smallImageKey = "autocode";
+
+			if (extension && supportedLanguages.includes(extension))
+				presenceData.largeImageKey = `lang-${extension}`;
+			else presenceData.largeImageKey = "autocode";
+		}
 	}
 
-	if (presenceData.details) presence.setActivity(presenceData);
+	if (!timestamp) delete presenceData.startTimestamp;
+
+	if (presenceData.smallImageKey) presence.setActivity(presenceData);
```

Comment:
if smallimagekey?

Reply:
actually I'll just make it if details or state